src/fu-engine.c

```diff
@@ -7307,6 +7320,18 @@ fu_engine_update_history_device(FuEngine *self, FuDevice *dev_history, GError **
 		}
 	}
 
+	/* check nothing changed */
```

Comment:
this isn't checking nothing changed, it's just re-measuring.

Reply:
Fixed.